1305.py

```diff
@@ -0,0 +1,35 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class Solution:
+    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:
+        def pushleft(node: TreeNode, stack: List[TreeNode]):
+            while node:
+                stack.append(node)
+                node = node.left
+
+        def addToRes(stack: List[TreeNode], res: List[int]):
```

Comment:
I can see that you are trying to group these common actions into a method to improve modularity. However, the current design of this method mixed two responsibilities into a single method: 1) add value to `res` and 2) perform `pop` and `push` against `stack`. This kind of design is not recommended.

Reply:
So I should not do this?